Support new filtering options on Topology page

EsdrasVP opened the pull request:
This PR is able to:
- Add support to new filtering options in the Topology page
- Add Health State filter options for Physical Infrastructure Topology page
- Add the possibility to have multiple values to the same filter (as the Unknown/None case)
- Change filters and names font-family to Open Sans

![image](https://user-images.githubusercontent.com/7563089/44092077-c8867d04-9fa5-11e8-83bb-f282d8373b3c.png)



Depends on:
[kubernetes-ui/topology-graph#29](https://github.com/kubernetes-ui/topology-graph/pull/29)

EsdrasVP commented:
@miq-bot assign @himdel

himdel commented:
@EsdrasVP I think this looks good :+1:  (I still have to test properly though)

But, looks like the specs need fixing to expect that extra field.. :)

himdel commented:
cc @skateman

EsdrasVP commented:
> @EsdrasVP I think this looks good  (I still have to test properly though)
> 
> But, looks like the specs need fixing to expect that extra field.. :)

@himdel Thanks for the reminder, fixed them and now it should work fine.

EsdrasVP commented:
@miq-bot add_label ux/review

@terezanovotna Could you take a look at this too?

terezanovotna commented:
Hi @EsdrasVP, 

1. Can we make the icons black?  Both in the top area and the topology area. When I look at Physical Chassis that has red icon, it seems like something is wrong with it, it is just not happy way to use color coding in this way.
2. I do not think we need icons for the health state. For ex: Valid is now represented with a checkbox, but that checkbox is not present anywhere in the topology area, and therefore can be confusing for the user. I suggest using just color dots.
3. Are we using SansOpen font for all?

Here is a mockup: 
![43656722-d01d4ebe-9729-11e8-8320-c9da1dc8a58d](https://user-images.githubusercontent.com/36040135/43707976-9413701e-9969-11e8-9e69-50ee166bff13.png)

What do you think @EsdrasVP? Let me know if you have any questions.

himdel commented:
Except for what @terezanovotna said, LGTM :).

All the existing topologies still work with https://github.com/kubernetes-ui/topology-graph/pull/29 (+ the css changes in here).

Re icon colors: even the blue icons should be black. (The blue was introduced by openstack at some point and got copied over, it should not be blue in containers either.)

EsdrasVP commented:
Hi @terezanovotna, regarding these questions:

1. We can, actually. Initially I thought that, since Racks were green as the state "Valid" and Switches were orange as "Warning", there  was no problem in using this color.
2. I've put icons on health state filters so the filter bar could be consistent. Also I didn't separate them as a new category for the same reason, but if it's more legible that's ok for me. I just have to work on this because currenlty there is no division by type in this div.
3. Currenlty no, it's only set as sans-serif, but I changed to Open Sans both filter names and display names.

Is there something else you want to address?

EsdrasVP commented:
> Re icon colors: even the blue icons should be black. (The blue was introduced by openshift at some point and got copied over, it should not be blue in containers either.)

@terezanovotna @himdel So all components icons should be black?

terezanovotna commented:
Yes, all components icons should be black @EsdrasVP

douglasgabriel commented:
Hey guys, I would say that we got a good readability of our topology differentiating components also by colors and not only by the icon, so, making all components black we will go back to the chaos that was before, to distinguish kinds of components in a dense topology. In a previous conversation with @terezanovotna we discuss about not use only colors to communicate with the user, once some people has different perceptions of colors, so, to solve this problem, we have different icons, but even so, we also can improve the communication with those people that not have such perception issues. @terezanovotna, @himdel and @EsdrasVP what do you think?

P.S. please don't make all icons black :pray: :cry:

himdel commented:
I think we want the icons black precisely so that we can use color for state, and not end up with a rainbow cake.

But.. not my decision, staying out of that one. :)

Not that related to this PR anyway (just don't add any red ones here please, those do look like an error indicator).

douglasgabriel commented:
here is @himdel https://github.com/ManageIQ/manageiq-ui-classic/pull/3999

terezanovotna commented:
@douglasgabriel @EsdrasVP I had no problem with colored icons before we added Health State. As health state is communicated via colors, we cannot use color coding for two same purposes as it is confusing for the user. 

As we show the health state in the colors, it would be great to have the icons black.

EsdrasVP commented:
@terezanovotna Ok, I made this change on [ManageIQ/manageiq-ui-classic#4434](https://github.com/ManageIQ/manageiq-ui-classic/pull/4434) so it would not interfere the objective of this PR.

himdel commented:
Thanks @EsdrasVP .. so... looks like this could be ready?

So far, there's been no response from the kubernetes-topology-graph maintainer  .. are you in any particular rush to get this in soon?

(I'm thinking let's give it a week, and then I can fork&release a manageiq version of the package, might as well use the opportunity to go npm-only with kubernetes-topology-graph, but if you're in a rush...?)

terezanovotna commented:
@miq-bot remove_label ux/review
@miq-bot add_label ux/approved

himdel commented:
@EsdrasVP looks like as it is now, this is breaking the styling of filters in all the other topologies:

![other](https://user-images.githubusercontent.com/289743/44088552-099b5528-9fb3-11e8-9ebb-7383c7d90830.png)

himdel commented:
@EsdrasVP now that the change in kubernetes-topology-graph is released, can you please add this to make it use that new release? :)

```diff
+diff --git a/app/assets/javascripts/application.js b/app/assets/javascripts/application.js
index 7826cc9b9..a980426bd 100644
--- a/app/assets/javascripts/application.js
+++ b/app/assets/javascripts/application.js
@@ -32,7 +32,7 @@
 //= require bower_components/d3/d3
 //= require bower_components/c3/c3
 //= require bower_components/lodash/lodash
-//= require bower_components/kubernetes-topology-graph/dist/topology-graph
+//= require kubernetes-topology-graph/dist/topology-graph
 //= require ./miq_browser_detect
 //= require ./miq_application
 //= require ./miq_flash
diff --git a/app/assets/stylesheets/application.css b/app/assets/stylesheets/application.css
index cf860cbae..e735d5251 100644
--- a/app/assets/stylesheets/application.css
+++ b/app/assets/stylesheets/application.css
@@ -11,7 +11,7 @@
  *= require ./miq_tree
  *= require bower_components/codemirror/lib/codemirror
  *= require bower_components/codemirror/theme/eclipse
- *= require bower_components/kubernetes-topology-graph/dist/topology-graph
+ *= require kubernetes-topology-graph/dist/topology-graph
  *= require ./wizard
  *= require ./timeline
  *= require ./topology
diff --git a/bower.json b/bower.json
index f519d06d2..d01339ab6 100644
--- a/bower.json
+++ b/bower.json
@@ -32,7 +32,6 @@
     "jquery-ui": "jqueryui#~1.12.1",
     "jquery-ujs": "~1.2.2",
     "jquery.observe_field": "himdel/jquery.observe_field#~0.1.0",
-    "kubernetes-topology-graph": "~0.0.23",
     "lodash": "~3.10.0",
     "manageiq-ui-components": "bower-dev",
     "moment-duration-format": "~1.3.0",
diff --git a/package.json b/package.json
index 81b8b8486..f2f6f61b3 100644
--- a/package.json
+++ b/package.json
@@ -35,6 +35,7 @@
     "graphql": "~0.12.0",
     "history": "^4.7.2",
     "jquery": "~2.2.4",
+    "kubernetes-topology-graph": "~0.0.25",
     "lodash": "^4.17.4",
     "patternfly": "~3.54.1",
     "patternfly-react": "2.10.1",
```

EsdrasVP commented:
> @EsdrasVP looks like as it is now, this is breaking the styling of filters in all the other topologies:

@himdel I changed the padding to see if it fixes the problem. Could you check that?

himdel commented:
@EsdrasVP all the topologies look consistent now, thanks! :)

himdel commented:
@EsdrasVP maybe 2 little things you could fix?

  * `.filter-group` - that's a bit too generic a name, can you use `topology-filter-group` or something similar?
  * `.topology .legend label` has `cursor: pointer` - that's fine, but it now means the "Health state:" label also has `cursor: pointer`, which is misleading as it's not really clickable. Can you adjust so that "Health state" uses the proper cursor?
  * and we can't merge without https://github.com/ManageIQ/manageiq-ui-classic/pull/4396#issuecomment-412845976

EsdrasVP commented:
> @EsdrasVP maybe 2 little things you could fix?
> 
> .filter-group - that's a bit too generic a name, can you use topology-filter-group or something similar?
> .topology .legend label has cursor: pointer - that's fine, but it now means the "Health state:" label also has cursor: pointer, which is misleading as it's not really clickable. Can you adjust so that "Health state" uses the proper cursor?
> and we can't merge without #4396 (comment)

@himdel That's fine, I changed the `.filter-group` to `.topology-filter-group` and added the `cursor: pointer` to the Health State label. Also, I already made the modifications required on [#4396 (comment)](https://github.com/ManageIQ/manageiq-ui-classic/pull/4396#issuecomment-412845976).

himdel commented:
> @himdel That's fine, I changed the .filter-group to .topology-filter-group and added the cursor: pointer to the Health State label. Also, I already made the modifications required on #4396 (comment).

@EsdrasVP Perfect, thanks!

Except... if that's so, you may need to `git push` ;)

EsdrasVP commented:
> @EsdrasVP Perfect, thanks!
> 
> Except... if that's so, you may need to git push ;)

@himdel I already did, I just closed and reopened the PR in order to Travis run correctly.

himdel commented:
@esdrasvp please look at the files changed screen.. There is no change to package.json, and I still see css for filter-group.

EsdrasVP commented:
> @EsdrasVP please look at the files changed screen.. There is no change to package.json, and I still see css for filter-group.

Sorry for that, don't know why only the change made in `package.json` was made. Now the modifications are here.
